### PR TITLE
Add SBOM chatbot sample

### PR DESCRIPTION
**Pull request by bechbd**
This is the sample application that was used as part of the AWS On-Air podcast and demonstrate natural language querying and graph RAG using LlamaIndex

### CODE REVIEW

File: solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                "Your task is to take the text provided and extract up to the "
+                "{max_knowledge_triplets} most important concepts in "
+                "knowledge triplets in the form of (subject, predicate, object).\n"
+                "Triplets should be focused on entities such as people, companies, locations, and events.\n"
+                "All triplets should not include stopwords such as a, an, and, are, as, at, be, but, by, for, if, in, into, is, it, no, not, of, on, or, such, that, the, their, then, there, these, they, this, to, was, will and with.\n"
+                "For each predicate, simplify the action into a single verb or a 2 words.\n"
+                "---------------------\n"
+                "Example:"
+                "Text: Amazon is located in Seattle."
+                "Triplets:\n(amazon, is located in, seattle)\n"
```
Comment (halcoope) [inline solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py:86]: Your example here as "is" and "in", which according to the prompt above should not be included.

File: solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            logger.info("Loading KnowledgeGraphIndex from local")
+            storage_context = StorageContext.from_defaults(
+                persist_dir=PERSIST_DIR + "/kg",
+                graph_store=self.kg_graph_store,
+            )
+            self.kg_index = load_index_from_storage(storage_context)
+            logger.info("Loading KnowledgeGraphIndex from local complete")
+
+    def run_kgrag_answer_question(self, question: str) -> DisplayResult:
+        """Runs the KnwoledgeGraph RAG Q/A
```
Comment (halcoope) [inline solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py:121]: Typo

File: solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+        Args:
+            question (str): The question being asked
+
+        Returns:
+            DisplayResult: A DisplayResult of the response
+        """
+        response = self.kg_query_engine.query(question)
+        explaination = []
```
Comment (halcoope) [inline solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py:130]: Typo

File: solutions/sbom-chatbot/NaturalLanguageQuerying.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    NeptuneDatabasePropertyGraphStore,
+)
+from llama_index.core import PromptTemplate
+from llama_index.llms.bedrock import Bedrock
+
+logger = logging.getLogger(__name__)
+
+
+class NaturalLanguageQuerying:
+    """Perform Natrual Language to OC query generation using Bedrock and LlamaIndex"""
```
Comment (halcoope) [inline solutions/sbom-chatbot/NaturalLanguageQuerying.py:24]: Typo

File: solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                "---------------------\n"
+                "Text: {text}\n"
+                "Triplets:\n"
+            )
+            template: PromptTemplate = PromptTemplate(text)
+
+            self.kg_index = KnowledgeGraphIndex.from_documents(
+                documents,
+                storage_context=kg_storage_context,
+                max_triplets_per_chunk=5,
```
[the remark this answers is not in the document]
Comment (bechbd) [reply solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py]: Made this configurable and added to the .env example

File: solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+                "Triplets:\n"
+                "(Amazon, acquires, Whole Foods Market)\n"
+                "(whole foods market, acquired for, $42 per share)\n"
+                "---------------------\n"
+                "Text: {text}\n"
+                "Triplets:\n"
+            )
+            template: PromptTemplate = PromptTemplate(text)
+
+            self.kg_index = KnowledgeGraphIndex.from_documents(
```
[the remark this answers is not in the document]
Comment (bechbd) [reply solutions/sbom-chatbot/KnowledgeGraphEnhancedRAG.py:100]: Good call, added

File: solutions/sbom-chatbot/KnowledgeGraphRetrieval.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+from llama_index.core.llms import ChatMessage
+from llama_index.core import PropertyGraphIndex
+from llama_index.llms.bedrock import Bedrock
+
+logger = logging.getLogger(__name__)
+
+
+class KnowledgeGraphRetriever:
+    """This demonstrates how to run Knowledge Graph Retrieval queries using
+    the CypherTemplateRetriever and LlamaIndex
```
[the remark this answers is not in the document]
Comment (bechbd) [reply solutions/sbom-chatbot/KnowledgeGraphRetrieval.py]: Added additional information to comment to guide the user in the things that will need to be changed here.

**Review by halcoope — APPROVED** [review]

**Pull request merged**